src/integer_mod_q/modulus_polynomial_ring_zq/get.rs

```diff
@@ -9,12 +9,54 @@
 //! Implementations to get content of a
 //! [`ModulusPolynomialRingZq].
 
-use super::ModulusPolynomialRingZq;
-use flint_sys::fq::fq_ctx_struct;
+use crate::{integer::Z, integer_mod_q::ModulusPolynomialRingZq};
+use flint_sys::{fmpz::fmpz_set, fq::fq_ctx_struct};
 
 impl ModulusPolynomialRingZq {
     /// Returns the [`fq_ctx_struct`] of a modulus and is only used internally.
     pub(crate) fn get_fq_ctx_struct(&self) -> &fq_ctx_struct {
         self.modulus.as_ref()
     }
+
+    /// Returns the context integer as a [`Z`]
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::Z;
+    /// use qfall_math::integer_mod_q::ModulusPolynomialRingZq;
+    /// use std::str::FromStr;
+    ///
+    /// let modulus_ring = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();
+    ///
+    /// let modulus = modulus_ring.get_q();
+    ///
+    /// # let cmp_modulus = Z:from(17);
+    /// # assert_eq!(cmp_modulus, modulus);
```

Comment:
```suggestion
    /// let cmp_modulus = Z::from(17);
    /// assert_eq!(cmp_modulus, modulus);
```
I am not sure if "context integer" would really help me tbh. But this example should make it clear.